src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -623,6 +713,28 @@ export class FileService implements IFileService {
 		});
 	}
 
+	// TODO@ben, I extracted this from `resolveStreamContent` but I am not sure
+	// what it does in comparsion to the other encoding related functions...
+	public getEncoding2(resource: uri, options: IResolveContentOptions, detected: IMimeAndEncoding): string {
```

Comment:
This is just a way to prioritise all the different encodings that can be determined for a file (either by detecting the encoding, by settings or by user choice). There are some rules which encoding to pick in which case. I think the method can be `private`, not `public` and be renamed to something like `selectEncoding` or `pickEncoding` or `getPreferredEncoding`.

Reply:
check